### PR TITLE
feat(rna): adding default dark mode export

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* Added a properly typed for RN default dark mode override

#### Description of how you validated changes

https://user-images.githubusercontent.com/321279/203151219-f44ef358-c30d-4d10-ab5f-dddbd5f1d755.mp4

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/theme/types.ts [hunk 1]
```diff
@@ -43,8 +43,8 @@ export type Components<Output> = Record<string, object> & {
   textField?: ComponentTheme<TextFieldStyles, Output>;
 };
 
-export type ColorMode = 'light' | 'dark' | 'system';
-type Override = Omit<Theme, 'overrides'> & {
+export type ColorMode = 'light' | 'dark' | 'system' | null;
```
Comment (ioanabrooks) [inline packages/react-native/src/theme/types.ts:46]: When would this be null? we should default to 'system', not allow null imho.

**Review by calebpollman — APPROVED** [review]
🚢

File: packages/react-native/src/theme/types.ts (hunk 1 above)
Comment (dbanksdesign) [reply packages/react-native/src/theme/types.ts:46]: If they pass the return of `useColorScheme` which can be null I guess? That is why I had to add it in

Comment (dbanksdesign) [reply packages/react-native/src/theme/types.ts:46]: https://reactnative.dev/docs/usecolorscheme

**Review by wlee221 — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**